### ISSUE TITLE
Add permit2 approval

### ISSUE THREAD

**Issue by rndquu** [ubiquity/onboard.ubq.fi#29]
Right now this repository is a 2-step setup wizard:
- Step 1: create `ubiquity-os` config
- Step 2: approve permit2 contract to spend tokens

Since the `ubiquity-os` config is now created in https://github.com/ubiquity-os/ubiquity-os-plugin-installer we should refactor the code in this repository:
1. Remove `Step 1: create "ubiquity-os" config`
2. Allow partner to approve [permit2](https://etherscan.io/address/0x000000000022d473030f116ddee9f6b43ac78ba3) contract to spend funds on partner's behalf
3. Rename current repository to smth like `permit2-allowance.ubq.fi` and deploy the frontend
4. Add `permit2-allowance.ubq.fi` URL to the readme file of https://github.com/ubiquity-os-marketplace/text-conversation-rewards

**Comment by zugdev** [comment]
Can I refactor it entirely?

**Comment by rndquu** [comment]
> Can I refactor it entirely?

Yes

**Comment by zugdev** [comment]
This, post refactor, might fit best in `pay.ubq.fi` when you connect with a treasury address.

**Comment by 0x4007** [comment]
I think we should leave everything in separate codebases as much as possible to make maintenance simpler.

**Comment by zugdev** [comment]
Is there an easy way to get the current reward token for an org?

**Comment by rndquu** [comment]
> Is there an easy way to get the current reward token for an org?

https://github.com/ubiquity-os/.ubiquity-os/blob/63bd7e7747c92262caf0a41cadf4f5bb1acc1884/.github/.ubiquity-os.config.dev.yml#L46

If `erc20RewardToken` param is no set then default values are DAI for mainnet and WXDAI for gnosis.

**Comment by 0x4007** [comment]
@rndquu Are the settings intentional on this repo?

**Comment by rndquu** [comment]
> [@rndquu](https://github.com/rndquu) Are the settings intentional on this repo?

Settings are taken from [this](https://github.com/ubiquity/.ubiquity-os/blob/21e1f089847c152cb0bca2516ed0c0aed8325687/.github/.ubiquity-os.config.yml#L19-L29) part of the config. Those settings match the original vision we discussed, outside contributors can `/start` tasks only with `Priority: 1 (Normal)` and `Priority: 2 (Medium)` labels.

**Comment by 0x4007** [comment]
An observation was made: why did we split off the allowances to a separate webapp? https://github.com/ubiquity/onboard.ubq.fi/pull/33#issuecomment-2651915864

**Comment by rndquu** [comment]
> Caution
> 
> Could not fetch base commit for this pull request

@gentlementlegen Help, I don't fully understand what's wrong here. [This](https://github.com/ubiquity/permit2-allowance.ubq.fi/pull/1) cross-repository PR solves current issue and has the `Resolves: #X` text in the PR description so, as far as I understand, rewards should be generated but they're not.

**Comment by rndquu** [comment]
> An observation was made: why did we split off the allowances to a separate webapp? [#33 (comment)](https://github.com/ubiquity/onboard.ubq.fi/pull/33#issuecomment-2651915864)

Because we have a convention of "many small web UIs".

**Comment by 0x4007** [comment]
> Because we have a convention of "many small web UIs".

Yes and I suppose that the permit allowance only needs to handle approval where onboard/demo need to generate configs and such. 

It just seems that there's a bit of overlap with the apps.

**Comment by gentlementlegen** [comment]
@rndquu It seems to be an issue with the ReviewIncentivizer module, I am not very familiar with the code base of this capability. I think the problem is that it didn't use the proper base branch of the proper repository thus failed to create the diff to evaluate, I'll open a ticket.

**Comment by zugdev** [comment]
Can we regenerate this too? @gentlementlegen

**Comment by zugdev** [comment]
I've regenerated rewards cause it didn't previously.

**Comment by whilefoo** [comment]
It seems git attributes were not fetched for some reason: https://github.com/ubiquity-os-marketplace/text-conversation-rewards/actions/runs/13592885861/job/38003041661#step:3:2235

EDIT: it's because git attributes needs to be in the repo where the PR is, so in this case `permit2-allowance.ubq.fi` needs to have `.gitattributes`
